src/components/ha-menu.ts

```diff
@@ -1,9 +1,29 @@
 import { MdMenu } from "@material/web/menu/menu";
+import type { CloseMenuEvent } from "@material/web/menu/menu";
+import {
+  CloseReason,
+  KeydownCloseKey,
+} from "@material/web/menu/internal/controllers/shared";
 import { css } from "lit";
 import { customElement } from "lit/decorators";
+import type { HaMenuItem } from "./ha-menu-item";
 
 @customElement("ha-menu")
 export class HaMenu extends MdMenu {
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.addEventListener("close-menu", this._handleCloseMenu);
+  }
+
+  private _handleCloseMenu(ev: CloseMenuEvent) {
+    if (
+      ev.detail.reason.kind === CloseReason.KEYDOWN &&
+      ev.detail.reason.key === KeydownCloseKey.ESCAPE
+    )
+      return;
```

Comment:
Pure style here, but when the statement gets pushed to the next line, I think it's much easier to read with braces.